docs/general.md

```diff
@@ -70,9 +72,103 @@ There is no way of modifying heaps via Fat Pointers, they can only be used to re
 
 ## Far Calls vs Near Calls. CallFrames and Context
 
-Explain the difference between the two, what information `CallFrame`s and `Context`s hold. Why `zksolc` wraps every far call in a near call to return the `success` boolean (with example assembly).
+Far Calls are the equivalent of calls in the EVM, they are used to call external contracts. Near Calls are used to call internal functions within the same contract that is being executed.
+
+Contracts have their own unique `Context` which itself can hold multiple `CallFrame`s. `CallFrame`s are used to keep track of the current state of the contract being executed.
+
+#### When a Far Call is made, a new `Context` is created and pushed into the running `Context`s of the vm. `Context`s are composed of:
+
+- Contract `Address`
+- Caller `Address`
+- Code `Address`
+- Code Page
+- `Stack`
+- Running `CallFrame`s (created by Near Calls)
+- `Heap`
+- `AuxHeap`
+- `CalldataHeap`
+
+The amount of gas that can be allocated to a new `Context` is limited to 63/64 of the currently available gas in the running `Context`.
+
+**A new Near Call will inherit the properties of the current `CallFrame`, and make use of the `Stack` and `Heap`s of the running `Context`**.
+
+#### `CallFrame`s are composed of:
+
+- Available gas
+- Exemption handler
+- Stack Pointer
+- Program Counter
+
+### Far Call wrapping
+
+Let's look at the following solidity code:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract LibraryContract {
+    function someFunction() public {}
+}
+
+contract CallerContract {
+    LibraryContract libraryContract;
+
+    constructor(address _libraryContractAddress) {
+        libraryContract = LibraryContract(_libraryContractAddress);
+    }
+
+    function callNonReturningFunction() public {
+        libraryContract.someFunction();
+    }
+}
+```
+
+Here we are calling a function of an external contract, this should net us a `far_call` instruction on our compiled code.
+
+Compiling this code with `zksolc` gives us (part of) the following assembly:
```

Comment:
I think this could be interpreted as:
The code generated by zksolc is a part of this complete code
Its actually the other way around.
Maybe:
> This is part of the assembly compiled with zksolc